lib/PaymentRails/BalanceGateway.php

```diff
@@ -56,6 +56,8 @@ public function search($query)
             }, $response['balances']);
 
             return new ResourceCollection($response, $items, $pager);
+        } else if ($response['errors']){
+            throw new Exception\Standard($response['errors']);
         } else {
             throw new Exception\DownForMaintenance();
```

Comment:
So DownForMaintenance() will never been throw - maybe HTTP 500 go to DownForMaintenance but 400 show the concat error strign.

Reply:
also maybe have a handleBadRequest in base Gateway/Http class.. 

and put the error response handler there and add `extends BaseGateway` for each Gateway. that way maybe we don't need to update 10 files..

Reply:
Yes.
As long as the response has a non-empty "errors" array, all the error messages will be concatenated into one string and raised as a StandardException

StandardExcetion already existing, so I used it.
We could make another class of our own if that suits.